### PR TITLE
add Locked trait

### PR DESCRIPTION
**Pull request by austinkline**
Adds a new trait to moments called `Locked`, which is an important field that platforms need to use to filter out moments that can't be used in listings/can't be sent anywhere

I ran the following for tests but I'm not sure if this is sufficient so please let me know if there's anything else I need to do

```
# from repo root
cd lib/go/contracts
make generate
make test
```

```
➜  contracts git:(is-locked-trait) ✗ make test
go test ./...
ok  	github.com/dapperlabs/nba-smart-contracts/lib/go/contracts	0.122s
?   	github.com/dapperlabs/nba-smart-contracts/lib/go/contracts/internal/assets	[no test files]
```

### CODE REVIEW

**Comment by tpetrychyn** [comment]
Hello, thanks for contributing!

There are tests in the root directory of the project which can be executed by running `make test` (They were broken, I just fixed them here https://github.com/dapperlabs/nba-smart-contracts/commit/9dd6390c3496ff4da08a84a01798c4e8082146d3)

Let me know when you've had a chance to update your branch and verify the tests and I will be happy to get this deployed for you.

**Comment by austinkline** [comment]
> Hello, thanks for contributing!
> 
> There are tests in the root directory of the project which can be executed by running `make test` (They were broken, I just fixed them here [9dd6390](https://github.com/dapperlabs/nba-smart-contracts/commit/9dd6390c3496ff4da08a84a01798c4e8082146d3))
> 
> Let me know when you've had a chance to update your branch and verify the tests and I will be happy to get this deployed for you.

Thanks for taking a look! I've gone ahead and verified tests pass (had to do two small updates to get them working), and I also had to commit a change to `assets.go` which seemed pretty similar to the flow-nft repo. If anything else is needed give me a shout!

Here's the output from running `make ci`

```bash
➜  nba-smart-contracts git:(is-locked-trait) go clean -testcache
➜  nba-smart-contracts git:(is-locked-trait) make ci
/Applications/Xcode.app/Contents/Developer/usr/bin/make ci -C lib/go
/Applications/Xcode.app/Contents/Developer/usr/bin/make ci -C contracts
go generate
go mod tidy
git diff --exit-code
go test ./...
ok  	github.com/dapperlabs/nba-smart-contracts/lib/go/contracts	0.117s
?   	github.com/dapperlabs/nba-smart-contracts/lib/go/contracts/internal/assets	[no test files]
/Applications/Xcode.app/Contents/Developer/usr/bin/make ci -C templates
go generate
go mod tidy
git diff --exit-code
go test ./...
?   	github.com/dapperlabs/nba-smart-contracts/lib/go/templates	[no test files]
?   	github.com/dapperlabs/nba-smart-contracts/lib/go/templates/data	[no test files]
?   	github.com/dapperlabs/nba-smart-contracts/lib/go/templates/internal/assets	[no test files]
/Applications/Xcode.app/Contents/Developer/usr/bin/make ci -C test
go generate
go mod tidy
git diff --exit-code
go test ./...
ok  	github.com/dapperlabs/nba-smart-contracts/lib/go/test	14.872s
```

**Comment by austinkline** [comment]
hey @tpetrychyn checking in, any chance I can get some eyes on this this week?

File: contracts/TopShot.cdc [hunk 1]
```diff
@@ -844,7 +844,8 @@ pub contract TopShot: NonFungibleToken {
                     let traitDictionary: {String: AnyStruct} = {
                         "SeriesNumber": TopShot.getSetSeries(setID: self.data.setID),
                         "SetName": TopShot.getSetName(setID: self.data.setID),
-                        "SerialNumber": self.data.serialNumber
+                        "SerialNumber": self.data.serialNumber,
+                        "Locked": TopShotLocking.isLocked(nftRef: &self as! &NonFungibleToken.NFT)
```
Comment (rrrkren) [inline contracts/TopShot.cdc:848]: not sure if you want LockExpiry as well `TopShotLocking.getLockExpiry`

**Review by HouseOfHufflepuff — APPROVED** [review]

**Review by tpetrychyn — APPROVED** [review]

File: contracts/TopShot.cdc (hunk 1 above)
Comment (austinkline) [reply contracts/TopShot.cdc:848]: Hadn't thought of that! I could go either way. The only way that LockExpiry is useful is if you can mark it as a `Date` and allow users to range on it. As of now, that isn't something Flowty can do

**Comment by tpetrychyn** [comment]
@austinkline it appears the build step got stuck during the github webhook outage yesterday. Could you please send an empty commit to restart the PR checks?

`gc --allow-empty -m "retry checks"`

**Comment by austinkline** [comment]
> @austinkline it appears the build step got stuck during the github webhook outage yesterday. Could you please send an empty commit to restart the PR checks?
> 
> `gc --allow-empty -m "retry checks"`

Done!

**Review by joshuahannan — APPROVED** [review]

**Pull request merged**